src/xtopcom/xstate_sync/src/xstate_sync.cpp

```diff
@@ -401,22 +403,65 @@ void xtop_state_sync::process_trie(state_req & req, std::error_code & ec) {
         }
         req.trie_tasks.erase(hash);
     }
-    for (auto const & blob : req.units_response) {
-        xinfo("xtop_state_sync::process_trie unit id: %u, blob size: %zu, {%s}", req.id, blob.size(), symbol().c_str());
-        auto hash = process_unit_data(blob, ec_internal);
-        if (ec_internal) {
-            if (ec_internal != evm_common::error::make_error_code(evm_common::error::xerrc_t::trie_sync_not_requested) &&
-                ec_internal != evm_common::error::make_error_code(evm_common::error::xerrc_t::trie_sync_already_processed)) {
-                xwarn("xtop_state_sync::process_trie invalid state node: %s, %s %s", to_hex(hash).c_str(), ec.category().name(), ec.message().c_str());
-                ec = ec_internal;
-                return;
-            } else {
-                xwarn("xtop_state_sync::process_trie process_unit_data abnormal: %s, %s %s", to_hex(hash).c_str(), ec_internal.category().name(), ec_internal.message().c_str());
+    // for (auto const & blob : req.units_response) {
+    //     xinfo("xtop_state_sync::process_trie unit id: %u, blob size: %zu,version:%d {%s}", req.id, blob.size(), req.version, symbol().c_str());
+    //     auto hash = process_unit_data(blob, req.version, ec_internal);
+    //     if (ec_internal) {
+    //         if (ec_internal != evm_common::error::make_error_code(evm_common::error::xerrc_t::trie_sync_not_requested) &&
+    //             ec_internal != evm_common::error::make_error_code(evm_common::error::xerrc_t::trie_sync_already_processed)) {
+    //             xwarn("xtop_state_sync::process_trie invalid state node: %s, %s %s", to_hex(hash).c_str(), ec.category().name(), ec.message().c_str());
+    //             ec = ec_internal;
+    //             return;
+    //         } else {
+    //             xwarn("xtop_state_sync::process_trie process_unit_data abnormal: %s, %s %s", to_hex(hash).c_str(), ec_internal.category().name(), ec_internal.message().c_str());
+    //         }
+    //     }
+    //     m_unit_bytes_uncommitted += blob.size();
+    //     req.unit_tasks.erase(hash);
+    // }
+
+    size_t i = 0;
+    
+    if (req.units_response.size() > 0) {
+        xassert(req.unit_tasks.size() == req.units_response.size());
+        if (req.unit_tasks.size() == req.units_response.size()) {
```

Comment:
better to add `else`?
if the code somewhere makes sure the tasks and the responses are matched one by one, only assert is enough.
if the equation is ensured here, `else` is a must choice.